code/modules/assembly/holder.dm

```diff
@@ -179,12 +179,15 @@
 		var/turf/T = get_turf(src)
 		if(!T)
 			return FALSE
+		user.unEquip(src, TRUE, TRUE)
 		if(a_left)
 			a_left.holder = null
 			a_left.forceMove(T)
-		if(a_right)
+			user.put_in_active_hand(a_left)
+		if(a_right) // Right object is the secondary item, hence put in inactive hand
 			a_right.holder = null
 			a_right.forceMove(T)
+			user.put_in_inactive_hand(a_right)
```

Comment:
why not use `put_in_any_hand_if_possible`

Reply:
I want the hands used to create the holder match what hands are used during during disassembly. a_left is the original item and a_right is the one that is added to create an assembly holder. Using active and inactive hands here allow for the video in `Images of changes` where the items go back to precisely where they were before, even if you swap hands it will still be the same igniter-remote signaler assembly and not a remote signaler-igniter assembly.